scripts/mkstatic.py

```diff
@@ -1137,6 +1144,71 @@ def write(self, metadata_request, metadata_define, arguments):
                         if not var_standard_name in arguments[scheme_name][subroutine_name]:
                             continue
 
+                        # To assist debugging efforts, check if arrays have the correct size (ignore scalars for now)
+                        assign_test = ''
+                        if debug:
+                            if ccpp_stage in ['init', 'timestep_init', 'timestep_finalize', 'finalize'] and \
+                                    CCPP_INTERNAL_VARIABLES[CCPP_BLOCK_NUMBER] in local_vars[var_standard_name]['name'] and \
+                                    '{}:{}'.format(CCPP_CONSTANT_ONE,CCPP_HORIZONTAL_DIMENSION) in var.dimensions:
+                                # We don't need extra tests for blocked arrays, because the de-blocking logic below
+                                # will catch any out of bound reads with the appropriate compiler flags. It naturally
+                                # deals with non-uniform block sizes etc.
+                                pass
+                            elif var.rank:
+                                array_size = []
+                                for dim in var.dimensions:
+                                    # This is not supported/implemented: tmpvar would have one dimension less
+                                    # than the original array, and the metadata requesting the variable would
+                                    # not pass the initial test that host model variables and scheme variables
+                                    # have the same rank.
+                                    if dim == CCPP_BLOCK_NUMBER:
+                                        raise Exception("{} cannot be part of the dimensions of variable {}".format(
+                                                                              CCPP_BLOCK_NUMBER, var_standard_name))
+                                    else:
+                                        # Handle dimensions like "A:B", "A:3", "-1:Z"
+                                        if ':' in dim:
+                                            dims = [ x.lower() for x in dim.split(':')]
+                                            try:
+                                                dim0 = int(dims[0])
+                                                dim0 = dims[0]
+                                            except ValueError:
+                                                if not dims[0].lower() in metadata_define.keys():
```

Comment:
Isn't `dims[0]` already lowercase based on call to `lower()` on line 1170? If so, this applies to the `lower()` calls below.